include/revng/Yield/Internal/Instruction.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <limits>
+#include <string>
+
+#include "revng/ADT/SortedVector.h"
+#include "revng/Model/VerifyHelper.h"
+#include "revng/Support/MetaAddress.h"
+#include "revng/Support/MetaAddress/YAMLTraits.h"
+#include "revng/Yield/Internal/TagType.h"
+
+namespace yield {
+using ByteContainer = llvm::SmallVector<uint8_t, 16>;
+}
+
+/* TUPLE-TREE-YAML
+
+name: Instruction
+type: struct
+fields:
+  - name: Address
+    type: MetaAddress
```

Comment:
Can we have `Start`/`End`?

Reply:
Sure.

Reply:
Do we really need `End` though? Consider that instructions are always stored in a vector and are considered packed. As such the `End` of one instructions is just the `Start` of the next one. The `End` of the last instruction in a basic block is the `End` of the corresponding block.

Reply:
Works, but then state this somewhere in the doc and check this in verify.

Reply:
[Done](https://github.com/Cvelth/revng/blob/snapshot/wip_cfg_in_svg/lib/Yield/Verify.cpp#L53).